sshd-core/src/main/java/org/apache/sshd/client/channel/ChannelSession.java

```diff
@@ -217,4 +223,42 @@ protected int securedRead(
             }
         }
     }
+
+    /**
+     * @param  key   The (never {@code null}) key (Note: may be empty...)
+     * @param  value The value to set - if {@code null} then the pre-existing value for the key (if any) is
+     *               <U>removed</U>.
+     * @return       The replaced/removed previous value - {@code null} if no previous value set for the key.
+     */
+    public Object setEnv(String key, Object value) {
+        ValidateUtils.checkNotNull(key, "No key provided");
```

Comment:
Please use `ValidateUtils#checkNotNullAndNotEmpty(key)`

Reply:
According to the current javadoc: key can be empty string. I am not planning to make any constraints changes to avoid behavior changes. 
@lgoldstein do you think we can change javadoc and update validation or use old validation without empty check?

Reply:
@tomaswolf Agree. Changes were reverted.